components/camel-iec60870/src/main/java/org/apache/camel/component/iec60870/BaseOptions.java

```diff
@@ -0,0 +1,223 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.iec60870;
+
+import java.util.Objects;
+import java.util.TimeZone;
+
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriParams;
+import org.eclipse.neoscada.protocol.iec60870.ASDUAddressType;
+import org.eclipse.neoscada.protocol.iec60870.CauseOfTransmissionType;
+import org.eclipse.neoscada.protocol.iec60870.InformationObjectAddressType;
+import org.eclipse.neoscada.protocol.iec60870.ProtocolOptions;
+
+@UriParams
+public abstract class BaseOptions<T extends BaseOptions<T>> {
+
+    /**
+     * Protocol options
+     */
+    @UriParam(javaType = "ProtocolOptions")
+    private ProtocolOptions.Builder protocolOptions;
+
+    // dummy for doc generation
+    /**
+     * The common ASDU address size.
+     * <p>
+     * May be either {@code SIZE_1} or {@code SIZE_2}.
+     * </p>
+     */
+    @UriParam(enums = "SIZE_1, SIZE_2", label = "connection")
+    private ASDUAddressType adsuAddressType;
+
+    // dummy for doc generation
```

Comment:
Actually I do need those fields only for generating the parameters in the documentation. The actual values are stored on the "options" object. That is why someone might not understand why the setters are not assigning to the fields.

Maybe there is a better way, but I would like to leave some marker about that in the file.

I am open to suggestions here!